File: Data Structure/card.java
import java.util.HashMap;
import java.util.Map;

public class Cart {
    private Map<String, Integer> products;

    public Cart() {
        products = new HashMap<>();
    }

    public void tambahProduk(String kodeProduk, int kuantitas) {
        if (products.containsKey(kodeProduk)) {
            int currentQuantity = products.get(kodeProduk);
            products.put(kodeProduk, currentQuantity + kuantitas);
        } else {
            products.put(kodeProduk, kuantitas);
        }
    }

    public void hapusProduk(String kodeProduk) {
        products.remove(kodeProduk);
    }

    public void tampilkanCart() {
        for (Map.Entry<String, Integer> entry : products.entrySet()) {
            System.out.println(entry.getKey() + " (" + entry.getValue() + ")");
        }
    }

    public static void main(String[] args) {
        Cart keranjang = new Cart();

        keranjang.tambahProduk("Pisang Hijau", 2);

        keranjang.tambahProduk("Semangka Kuning", 3);

        keranjang.tambahProduk("Apel Merah", 1);
        keranjang.tambahProduk("Apel Merah", 4);
        keranjang.tambahProduk("Apel Merah", 2);

        keranjang.hapusProduk("Semangka Kuning");

        keranjang.hapusProduk("Semangka Merah");

        keranjang.tampilkanCart();
    }
}

File: Data Structure/card.py
class Cart:
    def __init__(self):
        self.products = {}

    def tambahProduk(self, kodeProduk, kuantitas):
        if kodeProduk in self.products:
            currentQuantity = self.products[kodeProduk]
            self.products[kodeProduk] = currentQuantity + kuantitas
        else:
            self.products[kodeProduk] = kuantitas

    def hapusProduk(self, kodeProduk):
        if kodeProduk in self.products:
            del self.products[kodeProduk]

    def tampilkanCart(self):
        for kodeProduk, kuantitas in self.products.items():
            print(f"{kodeProduk} ({kuantitas})")

keranjang = Cart()

keranjang.tambahProduk("Pisang Hijau", 2)
keranjang.tambahProduk("Semangka Kuning", 3)
keranjang.tambahProduk("Apel Merah", 1)
keranjang.tambahProduk("Apel Merah", 4)
keranjang.tambahProduk("Apel Merah", 2)

keranjang.hapusProduk("Semangka Kuning")
keranjang.hapusProduk("Semangka Merah")

keranjang.tampilkanCart()
